code/game/objects/items/food/_food.dm

```diff
@@ -64,6 +66,8 @@
 	make_decompose(mapload)
 	make_bakeable()
 	make_microwaveable()
+	if(!starting_reagent_purity)
```

Comment:
should be if(starting_reagent_purity), this just automatically returns when it gets to the proc.